crates/telio-lana/src/lib.rs

```diff
@@ -115,3 +132,207 @@ pub fn deinit_lana() -> Result<moose::Result, moose::Error> {
 pub fn is_lana_initialized() -> bool {
     MOOSE_INITIALIZED.load(DEFAULT_ORDERING)
 }
+
+/// Initialize different moose variations (real, file or mock)
+///
+/// # Parameters:
+/// * event_path - path of the DB file where events will be stored. If such file does not exist, it will be created, otherwise reused.
+/// * app_version - Indicates the semantic version of the application.
+/// * prod - whether the events should be sent to production or not
+/// * tx - channel tx half for the InitCallback instance
+pub fn init_moose(
+    event_path: String,
+    app_version: String,
+    prod: bool,
+    tx: SyncSender<bool>,
+) -> std::result::Result<moose::Result, moose::Error> {
+    cfg_if! {
+        if #[cfg(test)] {
+            if let Some(mock) = test::MOCK.lock().unwrap().as_ref() {
+                return mock.init(
+                    event_path,
+                    LANA_APP_NAME.to_string(),
+                    app_version,
+                    LANA_MOOSE_VERSION.to_string(),
+                    prod,
+                    Box::new(moose_callbacks::MooseInitCallback {
+                        init_success_tx: tx,
+                    }),
+                    Box::new(moose_callbacks::MooseErrorCallback),
+                )
+            }
+        }
+    }
+    moose::init(
+        event_path,
+        LANA_APP_NAME.to_string(),
+        app_version,
+        LANA_MOOSE_VERSION.to_string(),
+        prod,
+        Box::new(moose_callbacks::MooseInitCallback {
+            init_success_tx: tx,
+        }),
+        Box::new(moose_callbacks::MooseErrorCallback),
+    )
+}
+#[cfg(test)]
+mod test {
+    use serial_test::serial;
+    use std::sync::Mutex;
+
+    use super::{
+        deinit_lana,
+        event_log::moose,
+        init_lana, is_lana_initialized,
+        moose::{ErrorCallback, InitCallback},
+    };
+
+    pub static MOCK: Mutex<Option<MooseStub>> = Mutex::new(None);
```

Comment:
Now that we use the Stub suffix this global should also be renamed to STUB

Reply:
Changed